agent/lib/broker_address_settings.js

```diff
@@ -31,52 +43,100 @@ function extract_address_plan(object) {
-    this.last = undefined;
+    this.last = {};
+    this.get_global_max_size().then(function (size) {
+        self.global_max_size = size;
+    });
 }
 
-BrokerAddressSettings.prototype.wait_for_plans = function () {
-    var self = this;
-    return new Promise(function (resolve, reject) {
-        self.watcher.once('updated', function () {
-            log.info('plans have been retrieved');
+BrokerAddressSettings.prototype.get_changes = function (name, addressplans, unchanged) {
+    var c = myutils.changes(this.last[name], addressplans, addressplan_compare, unchanged, description);
+    this.last[name] = addresses;
+    return c;
+};
+
+BrokerAddressSettings.prototype.get_global_max_size = function () {
+    var options = this.config;
+    var address_space_plan_path = kubernetes.get_path('/apis/admin.enmasse.io/v1beta2/namespaces/', 'addressspaceplans/' + this.config.ADDRESS_SPACE_PLAN, options);
+    return kubernetes.get_raw(address_space_plan_path, options).then(function (address_space_plan) {
+        if (address_space_plan.spec !== undefined) {
+            return address_space_plan.spec.infraConfigRef;
+        } else {
+            return 'default';
+        }
+    }).then(function (infraConfig) {
+        var infra_config_path = kubernetes.get_path('/apis/admin.enmasse.io/v1beta1/namespaces/', this.config.ADDRESS_SPACE_TYPE + 'infraconfigs/' + infraConfig, options);
+        return kubernetes.get_raw(infra_config_path, options).then(function (infra_config) {
+            var global_max;
+            if (infra_config.spec.broker.globalMaxSize) {
+                global_max = parseInt(infra_config.spec.broker.globalMaxSize);
+            }
+            return global_max;
         });
     });
-}
+};
+
+BrokerAddressSettings.prototype.dispatch = function (name, addressplans, description) {
+    log.info('dispatching: %s: %s', name, description);
+    this.emit(name, addressplans);
+};
 
 BrokerAddressSettings.prototype.update_settings = function (plan) {
     var r = required_broker_resource(plan);
     if (r && r > 0 && r < 1) {
-        this.required_broker_resource[plan.metadata.name] = r;
-        log.info('updated required broker resource for %s: %d', plan.metadata.name, this.required_broker_resource[plan.metadata.name]);
+        this.required_broker_resource[plan.name] = r;
+        log.info('updated required broker resource for %s: %d', plan.name, this.required_broker_resource[plan.name]);
     }
 };
 
 BrokerAddressSettings.prototype.generate_address_settings = function (plan, global_max_size) {
-    if (global_max_size) {
-        var r = this.required_broker_resource[plan];
-        if (r) {
-            return {
-                maxSizeBytes: r * global_max_size,
-                addressFullMessagePolicy: 'FAIL'
-            };
-        } else {
-            log.info('no broker resource required for %s, therefore not applying address settings', plan);
-        }
+    var r = this.required_broker_resource[plan];
```

Comment:
Doesn't this calculation need to factor in partitions?

Reply:
@lulf @vbusch if you have a plan that requires 0.5 broker and you are partitioning across 2, I think that means you should have 0.25 per broker.  I therefore think the calculation needs to divide by patitions.  Is that right?

Reply:
@lulf @vbusch is my comment invalid?

Reply:
Sorry, I have forgotten to respond to this. Yes, the credits should be divided by the number of partitions.